### DISCUSSION TITLE
Error when update PHP

### DISCUSSION THREAD

**Discussion by rickd1800** [netsuitephp/netsuite-php#144] [Q&A] [answered]
Hello All :)

We have been using this since 2019 and did a PHP update over the weekend. Not sure if this is related but now getting a critical error on the frontend of the webpage.

The developer who set this up is no longer so looking for some help on how to get this working again?

PHP Fatal error:  Uncaught Error: Undefined constant "NetSuite\SOAP_SINGLE_ELEMENT_ARRAYS" in /home/thermogroupcom/public_html/api/net-suite/vendor/ryanwinchester/netsuite-php/src/NetSuiteClient.php:114

**Comment by jrebs** [comment]
Looking at that line of that file, I am thinking you're using either a very different version from what is current or, possibly, a modified version that has its own changes. Would need to see more to be able to help.

**Comment by rickd1800** [comment]
Hello Justin :)

It is the 2019 version so much be quite different to current. 

This is the section it is from or do you want the whole file?
`    private function createOptions($config, $overrides = array())
    {
        return array_merge(array(
            'classmap' => require __DIR__."/includes/classmap.php",
            'trace' => 1,
            'connection_timeout' => 5,
            'location' => $config['host']."/services/NetSuitePort_".$config['endpoint'],
            'keep_alive' => false,
            'cache_wsdl' => WSDL_CACHE_BOTH,
            'features' => SOAP_SINGLE_ELEMENT_ARRAYS,
            'user_agent' => "PHP-SOAP/".phpversion()." + ryanwinchester/netsuite-php",
        ), $overrides);
    }`

**Comment by jrebs** [comment, marked as answer]
Perhaps you still need to install the php-soap extension.

**Comment by rickd1800** [comment]
Thankyou. I have tried that now however getting the below now:

PHP Fatal error:  Uncaught Error: Undefined constant "NetSuite\WSDL_CACHE_BOTH" in /home/thermogroupcom/public_html/api/net-suite/vendor/ryanwinchester/netsuite-php/src/NetSuiteClient.php:114

Do you have someone to recommend that I could get to have a look at the site and files?

**Reply by jrebs** [reply]
> Thankyou. I have tried that now however getting the below now:
> 
> PHP Fatal error: Uncaught Error: Undefined constant "NetSuite\WSDL_CACHE_BOTH" in /home/thermogroupcom/public_html/api/net-suite/vendor/ryanwinchester/netsuite-php/src/NetSuiteClient.php:114
> 
> Do you have someone to recommend that I could get to have a look at the site and files?

Unfortunately not. The error message you cite above is basically the same as the one in the earlier message. It's a different constant variable being fussed about, but it's still an issue where PHP seems to not be aware of the WSDL_CACHE_BOTH constant, which is provided by PHP when the SOAP extension is installed and being loaded: https://www.php.net/manual/en/soap.constants.php

Since the code is not aware of that constant existing, it is assuming it must be a constant in the NetSuite namespace and giving that error.  Before getting any deeper into things, I'd strongly recommend you get verification that the php-soap module is installed and being loaded by the same PHP process that is running the code in question. For instance, it's possible to have different configurations between CLI PHP and an libapache module. In the code that is failing, you should do a phpinfo or similar to see loaded extensions.

**Reply by rickd1800** [reply]
Thanks for your help. I have reinstalled the lastest package and it has solved the issue. Thanks for your help.

**Comment by MichaelBrenden** [comment]
We had hit similar, and it was a maddening oversight --

Using https://<acctnum>.connect.api.netsuite.com gave us similar 'missing WSDL' error.

Changed to https://<acctnum>.suitetalk.api.netsuite.com and immediate joy (and a moment of self-loathing for silly oversight: "suitetalk" instead of "connect").
